lib/chef/node/immutable_collections.rb

```diff
@@ -168,6 +174,12 @@ def to_h
 
       alias_method :to_hash, :to_h
 
+      # As Psych module, not respecting ImmutableMash object
+      # So first convert it to Hash/Array then parse it to `.to_yaml`
+      def to_yaml(*opts)
+        to_h.to_yaml(*opts)
+      end
```

Comment:
should probably add this API for the VividMash/ArrayAttr classes as well, since users can get at those out of the components.  i would think we'd need to add something in Chef::Node::Attribute as well to make this change visible.